examples/bigquery_view_parser.py

```diff
@@ -67,106 +67,112 @@ def _get_parser(cls):
+            LOOKUP, MERGE, NATURAL, NEW, NO, NOT, NULL, NULLS, OF, ON, OR, ORDER, OUTER, OVER,
+            PARTITION, PIVOT, PRECEDING, PROTO, RANGE, RECURSIVE, RESPECT, RIGHT, ROLLUP, ROWS,
+            SELECT, SET, SOME, STRUCT, TABLESAMPLE, THEN, TO, TREAT, TRUE, UNBOUNDED, UNION,
+            UNNEST, USING, WHEN, WHERE, WINDOW, WITH, WITHIN))
 
-        keyword = keyword_nonfunctions | MatchFirst(
-            (ESCAPE, CURRENT_TIME, CURRENT_DATE, CURRENT_TIMESTAMP, DATE_ADD,
-             DATE_SUB, ADDDATE, SUBDATE, INTERVAL, STRING_AGG, REGEXP_EXTRACT,
-             SPLIT, ORDINAL, UNNEST, SAFE_CAST, PARTITION, TIMESTAMP_ADD,
-             TIMESTAMP_SUB, ARRAY, GENERATE_ARRAY, GENERATE_DATE_ARRAY,
-             GENERATE_TIMESTAMP_ARRAY,
-             )
-        )
-        # fmt: on
+        # those are keywords that are fiunction names
+        keyword_funcs = MatchFirst((
+            TREAT, WITHIN, TABLESAMPLE, SOME, SET, RIGHT, PROTO, MERGE,
+            LOOKUP, LEFT, LATERAL, FETCH, CURRENT
+        ))
 
-        identifier_word = Word(alphas + "_@#", alphanums + "@$#_")
+        identifier_word = Word(alphas + '_@#', alphanums + '@$#_')
         identifier = ~keyword + identifier_word.copy()
         collation_name = identifier.copy()
-        # NOTE: Column names can be keywords.  Doc says they cannot, but in practice it seems to work.
-        column_name = identifier_word.copy()
-        qualified_column_name = Combine(
-            column_name + ("." + column_name)[..., 6], adjacent=False
-        )
+        # NOTE: Column names can't be keywords unless they are quoted
+        column_name = identifier.copy() | Suppress('`') + identifier_word + Suppress('`')
+        # first part of multi part column name can't be keyword, other parts can
+        qualified_column_name = Combine(column_name + (
+            ZeroOrMore(' ') + '.' + ZeroOrMore(' ') + identifier_word) * (0, 6))
+        qualified_column_name = qualified_column_name | Suppress('`') + qualified_column_name + Suppress('`')
         # NOTE: As with column names, column aliases can be keywords, e.g. functions like `current_time`.  Other
         # keywords, e.g. `from` make parsing pretty difficult (e.g. "SELECT a from from b" is confusing.)
-        column_alias = ~keyword_nonfunctions + column_name.copy()
+        # We will specifically exclude `from`, since we need to support trailing commas in the SELECT list, and
+        # SQL like `SELECT a, FROM b` becomes ambiguous if we support `from`.  In that SQL, is `FROM` a column name with
+        # alias `b`, or are we selecting a single column from table `b`?
+        column_alias = identifier.copy()
         table_name = identifier.copy()
         table_alias = identifier.copy()
         index_name = identifier.copy()
-        function_name = identifier.copy()
+
+        standard_name_part = ~keyword + Word(alphanums + "_" + "-") | keyword_funcs
+        quoted_name_part = Suppress("`") + CharsNotIn("`") + Suppress("`")
```

Comment:
Probably good to define a QuotedString("\`") expression and reuse it in place of `Suppress+CharsNotIn+Suppress`